consensus/transactionpool.go

```diff
@@ -164,3 +164,30 @@ func (s *State) TransactionPoolDump() (transactions []Transaction) {
 	defer s.mu.RUnlock()
 	return s.transactionPoolDump()
 }
+
+// State.AcceptTransaction() checks for a conflict of the transaction with the
+// transaction pool, then checks that the transaction is valid given the
+// current state, then adds the transaction to the transaction pool.
+// AcceptTransaction() is thread safe, and can be called concurrently.
+func (s *State) AcceptTransaction(t Transaction) (err error) {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+
+	// Check that the transaction is not in conflict with the transaction
+	// pool.
+	if s.transactionPoolConflict(&t) {
+		err = ConflictingTransactionErr
+		return
+	}
+
+	// Check that the transaction is potentially valid.
+	err = s.validTransaction(t)
+	if err != nil {
+		return
+	}
+
+	// Add the transaction to the pool.
+	s.addTransactionToPool(&t)
+
+	return
+}
```

Comment:
this...this is a good function. Only thing that sticks out is `s.validTransaction` taking `t` instead of `&t`.